dangerzone/document.py

```diff
@@ -18,13 +24,27 @@ class Document:
     document, and validating its info.
     """
 
-    def __init__(self, input_filename: str = None) -> None:
+    # document conversion state
+    STATE_UNCONVERTED = enum.auto()
+    STATE_CONVERTING = enum.auto()
+    STATE_SAFE = enum.auto()
+    STATE_FAILED = enum.auto()
+
+    def __init__(self, input_filename: str = None, output_filename: str = None) -> None:
+        self.id = uuid.uuid4()
```

Comment:
I was about to suggest using [`secrets.token_urlsafe(6)`](https://docs.python.org/3/library/secrets.html#secrets.token_urlsafe), which encodes random values as base64 strings, instead of hex strings which UUID4 does. This would encode more entropy within the same number of characters (36 bits of entropy vs 24 bits of entropy, for 6 characters).

However, I ran it through a simulation (you can play with it [here](https://www.wolframalpha.com/input?i2d=true&i=plot%5C%2840%29%5C%2840%291+-+Power%5Be%2CDivide%5B-k+*+%5C%2840%29k+-+1%5C%2841%29%2C2+*+Power%5B2%2C24%5D%5D%5D%5C%2841%29+*+100%5C%2844%29+0+%3C+k+%3C+1000%5C%2841%29)) based on [this article](https://preshing.com/20110504/hash-collision-probabilities/), and it seems that the chance of collision with 24 bits of entropy is low (0.03% for 100 documents, 3% for 1000 documents).

![collision](https://user-images.githubusercontent.com/2507626/200323575-9fe39add-2967-4b6e-925d-4421f9b6f14e.png)

Of course, with 36 bits of entropy, the chance of collision is virtually non-existent, less than 0.0007% for 1000 documents.

![collision2](https://user-images.githubusercontent.com/2507626/200326623-80c5146b-8eb4-4e74-b55a-06b23d59e0cb.png)

Anyway, the math say that we don't risk that much with UUID4, so we can go with them, if we prefer them to base64. I leave that to you, as I don't have a strong opinion, and this is not a critical property.

Reply:
I had assumed such a thing could happen (though I didn't looking into the probability) and I kept the full UUID as the full ID, but I guess from the user's perspective all they will see is the short version. So, for practical reasons we should avoid the collision there. So I have now done as you suggested.